### PR TITLE
Document behavior of `@QuarkusIntegrationTest` w/ container networks

### LINKED ISSUE #25010: Document limitation of `@QuarkusIntegrationTest` using Docker image
### Describe the bug

`@QuarkusIntegrationTest` runs against "any" "image type" (fast-jar, uber-jar, native image, Docker container), which is really neat.

`@QuarkusIntegrationTest` against Docker containers, when the tested application runs inside a Docker container, has an important limitation compared to the now deprecated-for-removal `@NativeImageTest` option, which is probably worth documenting.

Since the tested application runs inside a Docker container, the tested application cannot access resources running on the Docker host. Integration tests that for example use testcontainers need to be updated to use the proper Docker-network hostname.

`OidcWiremockTestResource` does *not* work with `@QuarkusIntegrationTest` against Docker containers, because the Wiremock server is running in the JVM running the test, which cannot be accessed from the Docker container running the Quarkus application.

Unfortunately there's no system property which can be easily checked for the currently tested Quarkus image type (fast/uber jar, native, docker). Seems worth to have such a property - or better: a way to only limit/restrict tests, likely with a new `@DisabledOnNativeDockerImage` annotation similar to `@DisabledOnNativeImage` and `@DisabledOnIntegrationTest`? Or add a set of "disabled image types" to `@DisabledOnIntegrationTest`?

Testcontainers `JdbcDatabaseContainer` example:
Old way was to simply use `JdbcDatabaseContainer.getJdbcUrl()`, but now it requires:
```java
public class MyTestResourceLifecycleManager implements QuarkusTestResourceLifecycleManager, DevServicesContext.ContextAware {
  private Optional<String> containerNetworkId;

  @Override
  public void setIntegrationTestContext(DevServicesContext context) {
    containerNetworkId = context.containerNetworkId();
  }

  @Override
  public Map<String, String> start() {
    JdbcDatabaseContainer container = new PostgreSQLContainer ...
    
    String jdbcUrl = container.getJdbcUrl();

    // Replace hostname + port in the provided JDBC URL with the hostname of the Docker container
    // running PostgreSQL and the listening port.
    if (containerNetworkId.isPresent()) {
      String hostPort = container.getHost() + ':' + container.getMappedPort(POSTGRESQL_PORT);
      String networkHostPort =
          container.getCurrentContainerInfo().getConfig().getHostName() + ':' + POSTGRESQL_PORT;
      jdbcUrl = jdbcUrl.replace(hostPort, networkHostPort);
    }

    return ImmutableMap.of(
        "quarkus.datasource.username", container.getUsername(),
        "quarkus.datasource.password", container.getPassword(),
        "quarkus.datasource.jdbc.url", jdbcUrl);
  }
}
`` 


### Expected behavior

-

### Actual behavior

Documentation updates

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.8.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by snazy**
* Enhance text + example for `@QuarkusIntegrationTest` in `getting-started-testing.adoc`
* Add a note to `security-openid-connect.adoc`

Fixes #25010

### CODE REVIEW

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 1]
```diff
@@ -1231,18 +1231,28 @@ When `@QuarkusIntegrationTest` results in launching a container (because the app
 This means that `@QuarkusIntegrationTest` works out of the box with containers launched via xref:dev-services.adoc[Dev Services], but it also means that it enables using <<quarkus-test-resource,QuarkusTestLifecycleManager>> resources that launch additional containers.
 This can be achieved by having your `QuarkusTestLifecycleManager` implement `io.quarkus.test.common.DevServicesContext.ContextAware`. A simple example could be the following:
 
+The container running the resource to test against, for example PostgreSQL via testcontainers, is assigned an IP address from the container's network.
+Use the container's "public" IP from its network and the "unmapped" port number to connect to the service.
+The testcontainers library usually return connection strings without respecting the container network, so additional code is needed to provide Quarkus the "correct" connection string using the container's IP on the container networt and the _unmapped_ port number.
```
Comment (gsmet) [inline docs/src/main/asciidoc/getting-started-testing.adoc:1236]:
```suggestion
The Testcontainers library usually return connection strings without respecting the container network, so additional code is needed to provide Quarkus the "correct" connection string using the container's IP on the container network and the _unmapped_ port number.
```

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 2]
```diff
@@ -1231,18 +1231,28 @@ When `@QuarkusIntegrationTest` results in launching a container (because the app
 This means that `@QuarkusIntegrationTest` works out of the box with containers launched via xref:dev-services.adoc[Dev Services], but it also means that it enables using <<quarkus-test-resource,QuarkusTestLifecycleManager>> resources that launch additional containers.
 This can be achieved by having your `QuarkusTestLifecycleManager` implement `io.quarkus.test.common.DevServicesContext.ContextAware`. A simple example could be the following:
 
+The container running the resource to test against, for example PostgreSQL via testcontainers, is assigned an IP address from the container's network.
```
Comment (gsmet) [inline docs/src/main/asciidoc/getting-started-testing.adoc:1234]:
```suggestion
The container running the resource to test against, for example PostgreSQL via Testcontainers, is assigned an IP address from the container's network.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks. I suggested a few micro changes.

**Comment by snazy** [comment]
Gentle ping @gsmet

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks!

**Pull request merged**